server/order/order.go

```diff
@@ -25,17 +28,69 @@ func (oid OrderID) String() string {
 	return hex.EncodeToString(oid[:])
 }
 
+// Value implements the sql/driver.Valuer interface.
+func (oid OrderID) Value() (driver.Value, error) {
+	return oid[:], nil // []byte
+}
+
+// Scan implements the sql.Scanner interface.
+func (oid *OrderID) Scan(src interface{}) error {
+	switch src := src.(type) {
+	case []byte:
+		copy(oid[:], src)
+		return nil
+		//case string:
+		// case nil:
+		// 	*oid = nil
+		// 	return nil
+	}
+
+	return fmt.Errorf("cannot convert %T to OrderID", src)
+}
+
 // OrderType distinguishes the different kinds of orders (e.g. limit, market,
 // cancel).
 type OrderType uint8
 
 // The different OrderType values.
 const (
-	LimitOrderType OrderType = iota
+	UnknownOrderType OrderType = iota
+	LimitOrderType
```

Comment:
Adding `UnknownOrderType` as the first item in this enum breaks all the tests, but if this is the right place for it, now is a time to make the change.  I do like having the unknown value be the zero value, which makes bugs that end up with the zero value easier to catch.

Reply:
Resolved in https://github.com/decred/dcrdex/pull/48, to be merged shortly.